recipes/default.rb

```diff
@@ -44,10 +44,35 @@
   cmd += " --ruleset='#{r}'"
 end
 
+# This file is maintained because the list of rulesets is not readily accessible
+# in a ThreatStack agent install, and we want to re-run the registration
+# process when the ruleset list changes.
+file '/opt/threatstack/etc/active_rulesets.txt' do
+  content node['threatstack']['rulesets'].join("\n").concat("\n")
+  mode 0644
+  owner 'root'
+  group 'root'
+end
+
+# deleting this file allows cloudsight to be reconfigured after installation
+file '/opt/threatstack/cloudsight/config/.secret' do
+  action :nothing
+  subscribes :delete, 'file[/opt/threatstack/etc/active_rulesets.txt]', :immediately
+end
+
+# Only if we are about to reconfigure a running instance
+execute 'stop threatstack services' do
+  command '/usr/bin/cloudsight stop'
+  action :nothing
+  subscribes :run, 'file[/opt/threatstack/etc/active_rulesets.txt]', :immediately
+end
+
 execute 'cloudsight setup' do
   command cmd
   action :run
   ignore_failure node['threatstack']['ignore_failure']
-  sensitive true
```

Comment:
I rebased and merged to our repo to avoid force-pushing the PR branch. Contents should be the same but there is an extra commit. LMK if this is not good enough - I'll have to rebase and resubmit a PR from a different branch then.